### PR TITLE
Remove dead code.

### PR DESCRIPTION
**Pull request by FGtatsuro**
I found several dead code in neovim/api/common.py. This PR removes them.

Best regards.

### CODE REVIEW

File: neovim/api/common.py [hunk 1]
```diff
@@ -154,10 +150,6 @@ def __contains__(self, item):
         return item in self._fetch()
 
 
-def _identity(obj, session, method, kind):
-    return obj
```
Comment (justinmk) [inline neovim/api/common.py]: This is for debugging I think

Comment (FGtatsuro) [reply neovim/api/common.py]: Oh, sorry. Thus, I remove this commit.

Comment (FGtatsuro) [reply neovim/api/common.py]: Force updated.

File: neovim/api/common.py [hunk 2]
```diff
@@ -69,8 +69,8 @@ class RemoteMap(object):
     It is used to provide a dict-like API to vim variables and options.
     """
 
-    def __init__(self, obj, get_method, set_method=None, self_obj=None):
-        """Initialize a RemoteMap with session, getter/setter and self_obj."""
```
Comment (justinmk) [inline neovim/api/common.py]: I'm not sure if this is used or not, but it seems like there's no test coverage. If it is used a test would be nice.

Comment (FGtatsuro) [reply neovim/api/common.py]: Thank you for your comment. As you say, I should add a test covering this change. 
I'll add several tests in test/test_common.py.

Comment (justinmk) [reply neovim/api/common.py]: Just one test is enough. @bfredl would know better whether `self_obj` is still needed.

Comment (bfredl) [reply neovim/api/common.py]: `self_obj` looks completely dead to me.

Comment (FGtatsuro) [reply neovim/api/common.py]: @bfredl 

Thank you for your check!

I've checked existing test code(under `test` directory), and I have a question.

`RemoteSequence`/ `RemoteMap` isn't public, and their `__init__` methods(fixed by this PR) are called internally. On the other hand, it seems that current test code covers public API. 
I wonder whether it's better to also cover private APIs(In this case, `__init__`  of `RemoteSequence`/ `RemoteMap`). 

Best regards.

Comment (bfredl) [reply neovim/api/common.py]: I don't think tests of internal classes are needed, if both classes are already covered by public API, there is no need to add a test.

Comment (FGtatsuro) [reply neovim/api/common.py]: OK, Thank you or your comment!

**Pull request merged**